### PR TITLE
warnings and price automation missed edge cases

### PR DESCRIPTION
**Pull request by justinzhou93**
remove warnings for rust linter, found edge case bugs for price automation, fixed some weights accordingly

### CODE REVIEW

File: pallets/automation-price/src/weights.rs [hunk 1]
```diff
@@ -76,24 +76,24 @@ impl<T: frame_system::Config> WeightInfo for AutomationWeight<T> {
...
-			.saturating_add(T::DbWeight::get().writes(6 as Weight))
+			.saturating_add(T::DbWeight::get().reads(3 as Weight))
+			.saturating_add(T::DbWeight::get().writes(7 as Weight))
 	}
 	fn asset_price_update_extrinsic() -> Weight{
 		(220_000_000 as Weight)
-			.saturating_add(T::DbWeight::get().reads(3 as Weight))
-			.saturating_add(T::DbWeight::get().writes(1 as Weight))
+			.saturating_add(T::DbWeight::get().reads(6 as Weight))
+			.saturating_add(T::DbWeight::get().writes(3 as Weight))
```
Comment (justinzhou93) [inline pallets/automation-price/src/weights.rs:85]: didn't account for withdrawing fees

File: pallets/automation-price/src/weights.rs [hunk 2]
```diff
@@ -76,24 +76,24 @@ impl<T: frame_system::Config> WeightInfo for AutomationWeight<T> {
...
-			.saturating_add(T::DbWeight::get().writes(4 as Weight))
+			.saturating_add(T::DbWeight::get().reads(4 as Weight))
+			.saturating_add(T::DbWeight::get().writes(5 as Weight))
 	}
 	fn schedule_transfer_task_extrinsic() -> Weight{
 		(200_000_000 as Weight)
-			.saturating_add(T::DbWeight::get().reads(4 as Weight))
-			.saturating_add(T::DbWeight::get().writes(2 as Weight))
+			.saturating_add(T::DbWeight::get().reads(6 as Weight))
+			.saturating_add(T::DbWeight::get().writes(4 as Weight))
```
Comment (justinzhou93) [inline pallets/automation-price/src/weights.rs:97]: didn't account for withdrawing fees

**Review by justinzhou93 — COMMENTED** [review]
couple of weight shifts cause of some catches in this PR, and then didn't redo weights after swapping from single asset to 2 and then the +2r,+2w for checking and withdrawing fees

File: pallets/automation-price/src/weights.rs [hunk 3]
```diff
@@ -76,24 +76,24 @@ impl<T: frame_system::Config> WeightInfo for AutomationWeight<T> {
 	}
 	fn delete_asset_extrinsic() -> Weight{
 		(220_000_000 as Weight)
-			.saturating_add(T::DbWeight::get().reads(2 as Weight))
-			.saturating_add(T::DbWeight::get().writes(6 as Weight))
+			.saturating_add(T::DbWeight::get().reads(3 as Weight))
+			.saturating_add(T::DbWeight::get().writes(7 as Weight))
```
Comment (justinzhou93) [inline pallets/automation-price/src/weights.rs:80]: check and update numberOfAssets

File: pallets/automation-price/src/weights.rs [hunk 4]
```diff
@@ -60,14 +60,14 @@ impl<T: frame_system::Config> WeightInfo for AutomationWeight<T> {
...
+			.saturating_add((20_000_000 as Weight).saturating_mul(v as Weight))
+			.saturating_add(T::DbWeight::get().reads((4 as Weight).saturating_mul(v as Weight)))
+			.saturating_add(T::DbWeight::get().writes((5 as Weight).saturating_mul(v as Weight)))
 	}
 	fn update_asset_reset() -> Weight{
 		(200_000_000 as Weight)
-			.saturating_add(T::DbWeight::get().reads(1 as Weight))
-			.saturating_add(T::DbWeight::get().writes(2 as Weight))
+			.saturating_add(T::DbWeight::get().reads(2 as Weight))
+			.saturating_add(T::DbWeight::get().writes(1 as Weight))
```
Comment (justinzhou93) [inline pallets/automation-price/src/weights.rs:70]: got the number of reads and writes flipped and update_asset_reset is also used in reset_asset function

File: pallets/automation-price/src/weights.rs [hunk 5]
```diff
@@ -76,24 +76,24 @@ impl<T: frame_system::Config> WeightInfo for AutomationWeight<T> {
...
+			.saturating_add(T::DbWeight::get().writes(3 as Weight))
 			.saturating_add(T::DbWeight::get().reads(21 as Weight))
 			.saturating_add(T::DbWeight::get().writes(21 as Weight))
 	}
 	fn add_asset_extrinsic() -> Weight{
 		(220_000_000 as Weight)
-			.saturating_add(T::DbWeight::get().reads(3 as Weight))
-			.saturating_add(T::DbWeight::get().writes(4 as Weight))
+			.saturating_add(T::DbWeight::get().reads(4 as Weight))
+			.saturating_add(T::DbWeight::get().writes(5 as Weight))
```
Comment (justinzhou93) [inline pallets/automation-price/src/weights.rs:92]: scheduled asset deletion change above, didn't add the 1r, 1w

**Review by rhuttman — APPROVED** [review]

**Pull request merged**